### PR TITLE
[EOSD-727] adding support for 1.15 and updating cni

### PR DESCRIPTION
**Pull request by shabir61**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -27,7 +27,7 @@ T_YELLOW := \e[0;33m
 T_RESET := \e[0m
 
 .PHONY: all
-all: 1.10 1.11 1.12
+all: 1.10 1.11 1.12 1.14
```
Comment (barleyj-opengov) [inline Makefile]: Should we add 1.15 since we want to upgrade to that at some point?

Comment (shabir61) [reply Makefile]: yes - working on to add 1.15

**Review by barleyj-opengov — DISMISSED** [review]

**Review by barleyj-opengov — DISMISSED** [review]

**Review by Chili-Man — CHANGES_REQUESTED** [review]
You need to update this repository with the upstream matser

File: README.md [hunk 2]
```diff
@@ -61,6 +61,8 @@ versions by running `aws s3 ls s3://amazon-eks/cloudformation/`.
 
 | CloudFormation Version | EKS AMI versions                           | [amazon-vpc-cni-k8s](https://github.com/aws/amazon-vpc-cni-k8s/releases) |
 | ---------------------- | ------------------------------------------ | -------------------- |
+| 2019-09-27             | amazon-eks-node-(1.14,1.13,1.12,1.11)-v20190927 | v1.5.4
```
Comment (barleyj-opengov) [inline README.md:64]: I don't see 1.15 in here. Should it be?

**Review by barleyj-opengov — COMMENTED** [review]
I'll need a bit more time to go through the CloudFormation template and the log collector

File: .gitignore [hunk 3]
```diff
@@ -1,2 +1,3 @@
 manifest.json
 *.swp
+.idea
```
Comment (sakomws) [inline .gitignore]: new line

File: files/1.14/kubelet.service [hunk 4]
```diff
@@ -0,0 +1,21 @@
+[Unit]
```
Comment (Chili-Man) [inline files/1.14/kubelet.service]: where did this directory and file come from? this is not in the master branch from upstream

File: log-collector-script/windows/eks-ssm-content.json [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+               "    Write-Host 'Uploaded to S3 failed' ",
+               "    break ",
+               "} ",
+               "",
+               "Remove-Item -Force .\\eks-log-collector.ps1 -ErrorAction Ignore "
+            ]     
+         }
+      }
+   ]
+}
```
Comment (sakomws) [inline log-collector-script/windows/eks-ssm-content.json]: new line

File: README.md (hunk 2 above)
Comment (shabir61) [reply README.md:64]: I am not sure - our jenkins tests build 1.15 -

File: .gitignore (hunk 3 above)
Comment (shabir61) [reply .gitignore]: really

File: log-collector-script/windows/eks-ssm-content.json (hunk 5 above)
Comment (shabir61) [reply log-collector-script/windows/eks-ssm-content.json]: what is that mean new line?

File: scripts/install-worker.sh [hunk 6]
```diff
@@ -76,6 +80,9 @@ sudo bash install -u false
 # Remove the agent installation script
 rm install
 
+# Remove the ec2-net-utils package, if it's installed. This package interferes with the route setup on the instance.
```
Comment (Chili-Man) [inline scripts/install-worker.sh:80]: this seems to be duplicated from lines 69-70 above; remove it from here

File: .gitignore (hunk 3 above)
Comment (Chili-Man) [reply .gitignore]: yes

File: scripts/install-worker.sh [hunk 7]
```diff
@@ -194,22 +201,21 @@ for binary in ${BINARIES[*]} ; do
...
-    KUBELET_CONFIG=kubelet-config-with-secret-polling.json
-fi
 
 sudo mkdir -p /etc/kubernetes/kubelet
 sudo mkdir -p /etc/systemd/system/kubelet.service.d
 sudo mv $TEMPLATE_DIR/kubelet-kubeconfig /var/lib/kubelet/kubeconfig
 sudo chown root:root /var/lib/kubelet/kubeconfig
-sudo mv $TEMPLATE_DIR/kubelet.service /etc/systemd/system/kubelet.service
+
+if [ "$KUBERNETES_MINOR_VERSION" = "1.14" ]; then
```
Comment (Chili-Man) [inline scripts/install-worker.sh]: I'm not sure where this logic is coming from; its not in the upstream master  anywhere: https://github.com/awslabs/amazon-eks-ami/blob/master/scripts/install-worker.sh ; you might want to make sure that you are pulling in the changes from the latest upstream master

File: Jenkinsfile [hunk 8]
```diff
@@ -3,9 +3,9 @@
 @Library('OGJenkinsLib@3.7.0') _
 
 final GIT_REPOSITORY_NAME = 'amazon-eks-ami'
-final KUBERNETES_VERSION = '1.13.7'
+final KUBERNETES_VERSION = '1.15.10'
```
Comment (Chili-Man) [inline Jenkinsfile:6]:
```suggestion
final KUBERNETES_VERSION = '1.15.11'
```
is the latest version available https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html

Comment (Chili-Man) [inline Jenkinsfile:6]: also Bump OGJenkinslib to 3.18  on line 3

File: README.md (hunk 2 above)
Comment (Chili-Man) [reply README.md:64]: this is how it is from upstream, so we'll leave it as is

File: eks-worker-al2.json [hunk 9]
```diff
@@ -1,35 +1,51 @@
...
 
-    "arch": "",
-    "instance_type": "",
-    "binary_bucket_name": "",
-    "docker_version": "",
+    "binary_bucket_name": "amazon-eks",
+    "binary_bucket_region": "us-west-2",
+    "kubernetes_version": null,
+    "kubernetes_build_date": null,
+    "docker_version": "18.09.9ce-2.amzn2",
```
Comment (Chili-Man) [inline eks-worker-al2.json]: have it use the latest version instead `19.03.6ce-1.amzn2`

File: eks-worker-al2.json [hunk 10]
```diff
@@ -1,35 +1,51 @@
...
+    "cni_version": "v0.6.0",
+    "cni_plugin_version": "v0.7.5",
```
Comment (Chili-Man) [inline eks-worker-al2.json]:
```suggestion
    "cni_plugin_version": "v0.7.6",
```

**Review by Chili-Man — CHANGES_REQUESTED** [review]

File: files/1.14/kubelet.service (hunk 4 above)
Comment (shabir61) [reply files/1.14/kubelet.service]: we dont need this now - removing it.

File: scripts/install-worker.sh (hunk 6 above)
Comment (shabir61) [reply scripts/install-worker.sh:80]: removed duplicates-

File: scripts/install-worker.sh (hunk 7 above)
Comment (shabir61) [reply scripts/install-worker.sh]: removed 1.14 patch

File: Jenkinsfile (hunk 8 above)
Comment (shabir61) [reply Jenkinsfile:6]: bumped to 11

Comment (shabir61) [reply Jenkinsfile:6]: bumped 3.18.0

File: eks-worker-al2.json (hunk 9 above)
Comment (shabir61) [reply eks-worker-al2.json]: bumped to 19.03.6ce-1.amzn2

File: eks-worker-al2.json (hunk 10 above)
Comment (shabir61) [reply eks-worker-al2.json]: bumped to v0.7.6

**Comment by shabir61** [comment]
revert back to 1.15.10 - i dont see 1.15.11 yet - 
```
aws-okta exec production -- aws s3 ls s3://amazon-eks
    echo "Account IDs to build images for: ${utils.jsonify(config.accountIds)}"
INFO[0001] Requesting MFA. Please complete two-factor authentication with your second device

source_ami_owners ?= 141808717104
Touch the flashing U2F device to authenticate...
  ==> Touch accepted. Proceeding with authentication
                           PRE 1.10.11/
                           PRE 1.10.13/
                           PRE 1.10.3/
                           PRE 1.11.10/
                           PRE 1.11.5/
                           PRE 1.11.8/
                           PRE 1.11.9/
                           PRE 1.12.10/
                           PRE 1.12.7/
                           PRE 1.12.9/
                           PRE 1.13.11/
                           PRE 1.13.12/
                           PRE 1.13.7/
                           PRE 1.13.8/
                           PRE 1.14.6/
                           PRE 1.14.7/
                           PRE 1.14.8/
                           PRE 1.14.9/
                           PRE 1.15.10/
                           PRE cloudformation/
                           PRE manifests/
```

File: log-collector-script/windows/eks-ssm-content.json (hunk 5 above)
Comment (sakomws) [reply log-collector-script/windows/eks-ssm-content.json]: Screen Shot 2020-04-17 at 6 07 43 PM

File: Jenkinsfile (hunk 8 above)
Comment (Chili-Man) [reply Jenkinsfile:6]: still says `.10`

**Review by Chili-Man — APPROVED** [review]

**Review by Chili-Man — CHANGES_REQUESTED** [review]
show me that the build works

**Review by Chili-Man — APPROVED** [review]

**Review by barleyj-opengov — APPROVED** [review]

**Pull request merged**